### PR TITLE
PROTO-3101: Add outlier resistant network inferences

### PR DESCRIPTION
**Pull request by KjetilVaa**
- Adds outlier resistant inference data to existing CRON job and stores it in `query_results` table

### CODE REVIEW

**Review by felipead — COMMENTED** [review]
Is it possible to add some unit testing?

File: app/usecase/queryproducer.go [hunk 1]
```diff
@@ -87,6 +87,20 @@ func (q *QueryProducer) ProcessQueries(ctx context.Context) error {
 		}
 	}
 
+	for i := uint64(1); i <= maxTopicID; i++ {
```
Comment (felipead) [inline app/usecase/queryproducer.go:90]: Can we cover this logic with unit or integration tests? It's very complex.

File: app/service/queryprocessor.go [hunk 2]
```diff
@@ -154,6 +154,36 @@ func (q *QueryProcessorService) ProcessEmissionInfo(ctx context.Context) error {
...
+	metadata := json.RawMessage(fmt.Sprintf(`{"block_height": %d, "topic_id": %d}`, latestAvailableNetworkInferences.Height, topicID))
+	err = q.repo.SaveQueryResult(ctx, fmt.Sprintf("latest_available_network_inferences_outlier_resistant_%d", topicID), "latest_available_network_inferences_outlier_resistant", metadata, latestAvailableNetworkInferences.Data)
+	if err != nil {
+		return fmt.Errorf("failed to save query result: %w", err)
+	}
+
+	return nil
+}
+
+func (q *QueryProcessorService) ProcessLatestNetworkInferencesOutlierResistant(ctx context.Context, topicID uint64) error {
```
Comment (felipead) [inline app/service/queryprocessor.go:172]: Some unit or integration tests in this method

File: app/service/queryprocessor.go [hunk 3]
```diff
@@ -154,6 +154,36 @@ func (q *QueryProcessorService) ProcessEmissionInfo(ctx context.Context) error {
 	return nil
 }
 
+func (q *QueryProcessorService) ProcessLatestAvailableNetworkInferencesOutlierResistant(ctx context.Context, topicID uint64) error {
```
Comment (felipead) [inline app/service/queryprocessor.go:157]: Some unit or integration tests in this method

**Review by fernandofcampos — APPROVED** [review]

**Comment by fernandofcampos** [comment]
Tests will be added at https://github.com/allora-network/allora-producer/pull/23 because it is refactoring to code to improve testability.

**Pull request merged**